### PR TITLE
[feature/ef-search -> main] Adds Nprobe as a query parameter in knn query

### PR DESCRIPTION
**Pull request by shatejas**
Adds integration test for invalid method parameters (#1782)

### Description
Adds nprobes as a query parameter

This is cherry-picked from feature branch to main. Original PRs
- https://github.com/opensearch-project/k-NN/pull/1758
- https://github.com/opensearch-project/k-NN/pull/1782

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 ### Features
 * Adds dynamic query parameter ef_search [#1783](https://github.com/opensearch-project/k-NN/pull/1783)
 * Adds dynamic query parameter ef_search in radial search faiss engine [#1790](https://github.com/opensearch-project/k-NN/pull/1790)
+* Adds dynamic query parameter nprobes [1792](https://github.com/opensearch-project/k-NN/pull/1792)
```
Comment (jmazanec15) [inline CHANGELOG.md:19]: nit: 1792 -> #1792

**Review by jmazanec15 — COMMENTED** [review]
How is check handled to see if mapping uses IVF?

File: CHANGELOG.md (hunk 1 above)
Comment (shatejas) [reply CHANGELOG.md:19]: will change

**Comment by shatejas** [comment]
> How is check handled to see if mapping uses IVF?

Happens in [doToQuery()](https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java#L528). It will throw an exception if there there is [no context found ](https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/index/util/AbstractKNNLibrary.java#L42)

**Comment by ryanbogan** [comment]
If I understand correctly, these changes are already in the feature branch?

**Comment by shatejas** [comment]
> If I understand correctly, these changes are already in the feature branch?

Thats correct. I have linked the original Prs in the description

**Review by ryanbogan — DISMISSED** [review]
LGTM!

**Review by jmazanec15 — APPROVED** [review]
LGTM thanks!

**Review by navneet1v — APPROVED** [review]

**Pull request merged**